arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/ParTraverse.kt

```diff
@@ -115,12 +150,34 @@ suspend fun <A, B> Iterable<A>.parTraverseN(n: Long, f: suspend (A) -> B): List<
  *
  * Cancelling this operation cancels all running tasks.
  */
+@Deprecated("use parTraverseN with n as Int type", ReplaceWith("parTraverseN", "arrow.fx.coroutines.parTraverseN"))
 suspend fun <A, B> Iterable<A>.parTraverseN(
   ctx: CoroutineContext = EmptyCoroutineContext,
   n: Long,
   f: suspend (A) -> B
 ): List<B> {
-  val s = Semaphore(n.toInt())
+  require(n <= Int.MAX_VALUE) { "n must be <= ${Int.MAX_VALUE}, was: $n" }
+  val s = Semaphore(Math.toIntExact(n))
```

Comment:
```suggestion
  val s = Semaphore(Math.toIntExact(n))
```

`Math.toIntExact` already checks for overflowing https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#toIntExact-long-.